src/nodes/chunk_dispatch.c

```diff
@@ -144,10 +146,67 @@ ts_chunk_dispatch_get_chunk_insert_state(ChunkDispatch *dispatch, Point *point,
 		 * locking the hypertable. This serves as a fast path for the usual case
 		 * where the chunk already exists.
 		 */
+		bool found;
 		Chunk *new_chunk = ts_hypertable_find_chunk_for_point(dispatch->hypertable, point);
 		if (new_chunk == NULL)
 		{
-			new_chunk = ts_hypertable_create_chunk_for_point(dispatch->hypertable, point);
+			new_chunk = ts_hypertable_create_chunk_for_point(dispatch->hypertable, point, &found);
+		}
+		else
+			found = true;
+
+		/* get the filtered list of "available" DNs for this chunk */
+		List *chunk_data_nodes =
```

Comment:
This part feels like doubles logic in the `get_chunk_data_nodes()`, maybe it can be reused

Reply:
`get_chunk_data_nodes` doesn't take into account available DNs and there are many other callers of it.